ui/templates/listing.html

```diff
@@ -0,0 +1,83 @@
+{% extends "base.html" %}
+{% block content %}
+<div id="header">
+<h1>Listing</h1>
+</div>
+
+<div>
+    <a href="{% url 'upload' repo.id %}">import</a>
+</div>
+
+<div id="nav" style="float: left; height: 100%;">
+<p>
+<h2>Repository {{ repo.name }} </h2>
+</p>
+
+{% if courses.count > 0 %}
```

Comment:
Just a suggestion, but it might be nice to roll all this into an include file so it is easier to see what is going on here with the logic.